src/System.AppContext/tests/AppContext.BaseDirectory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Diagnostics;
+using Xunit;
+
+namespace System.Tests
+{
+    public partial class AppContextTests
+    {
+        [Fact]
+        [ActiveIssue(6677, PlatformID.OSX)]
+        public void DefaultMatchesGetModuleFileName()
+        {
+            Assert.Equal(GetMainModuleDirectory(), AppContext.BaseDirectory.TrimEnd(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar));
+        }
+
+        private static string GetMainModuleDirectory()
```

Comment:
I know @AlexGhiondea was going to try and figure out how to correctly test this for the various platforms once we have the infrastructure in place.